src/Type/SchemaConfig.php

```diff
@@ -37,7 +37,7 @@ class SchemaConfig
     /** @var ObjectType|null */
     public $subscription;
 
-    /** @var Type[]|callable */
+    /** @var array<Type>|(callable(): array<Type>) */
```

Comment:
what is the array key type here?

Reply:
It is unspecified, because it does not matter. The index is only used for debugging, we don't have to constrain it unnecessarily.

Reply:
IMO we should use as much narrow type as possible. If index is always int, we should specify it. `array<Type>` is equal to `array<mixed, Type>` so the type of key is as narrow as possible.

Reply:
This property accepts an array given by users of this library, they can use whatever keys they want. It is not significant for our purposes.